### PR TITLE
Add MDS stats

### LINKED ISSUE #54: Support for mds stats
`mds.MDS.mdt.stats`:
```
mds.MDS.mdt.stats=
snapshot_time             1688128383.546447404 secs.nsecs
req_waittime              134066237 samples [usec] 2 223783 2088082648 471534475048
req_qdepth                134066237 samples [reqs] 0 33 360860 426556
req_active                134066237 samples [reqs] 1 82 174309144 308295414
req_timeout               134066237 samples [sec] 1 15 2010993249 30164896809
reqbuf_avail              274307261 samples [bufs] 61 64 17553178393 1123246805789
ldlm_ibits_enqueue        81452437 samples [reqs] 1 1 81452437 81452437
mds_reint_setattr         4377 samples [reqs] 1 1 4377 4377
mds_reint_create          17987068 samples [reqs] 1 1 17987068 17987068
mds_reint_unlink          34215148 samples [reqs] 1 1 34215148 34215148
mds_reint_open            33128857 samples [reqs] 1 1 33128857 33128857
ost_set_info              345 samples [usec] 7 50 7294 180944
mds_connect               73 samples [usec] 7 20470 30960 436899558
mds_disconnect            27 samples [usec] 39 393 5412 1514890
mds_get_root              28 samples [usec] 5 23 284 3574
mds_statfs                270 samples [usec] 6 52 5142 115274
mds_sync                  18256 samples [usec] 5 721 367130 11121902
obd_ping                  388208 samples [usec] 2 47039 6770169 21323667737
```

`mds.MDS.mdt_readpage.stats`:
```
mds.MDS.mdt_readpage.stats=
snapshot_time             1688128401.905932373 secs.nsecs
req_waittime              33128745 samples [usec] 2 35496 511938509 27521514737
req_qdepth                33128745 samples [reqs] 0 6 58069 63753
req_active                33128745 samples [reqs] 1 6 36206011 43141697
req_timeout               33128745 samples [sec] 15 15 496931175 7453967625
reqbuf_avail              67052468 samples [bufs] 61 64 4278031539 272954462355
mds_getattr               28 samples [usec] 26 58 949 33583
mds_close                 33128667 samples [usec] 8 130142 611180236 143780158340
mds_readpage              50 samples [usec] 118 1601 14411 6161215
```

### PR DESCRIPTION
**Pull request by RDruon**
Fix #54

### CODE REVIEW

File: src/stats_parser.rs [hunk 1]
```diff
@@ -108,7 +109,7 @@ where
     I: Stream<Token = char>,
     I::Error: ParseError<I::Token, I::Range, I::Position>,
 {
-    (newline().with(time_triple()), many(stat())).map(|(_, xs)| xs)
+    (optional(newline()).with(time_triple()), many(stat())).map(|(_, xs)| xs)
```
[the remark this answers is not in the document]
Comment (RDruon) [reply src/stats_parser.rs:112]: Due to https://github.com/whamcloud/lustre-collector/pull/60/files#diff-39f6a5faac36c84672903b2ec09ae44131b3e7994104c970f229bbeb34684cb0R289

File: src/mds/mds_parser.rs [hunk 2]
```diff
@@ -1,173 +1,100 @@
...
-    .collect()
+fn param_non_final<I>(x: &'static str) -> impl Parser<I, Output = Param>
+where
+    I: Stream<Token = char>,
+    I::Error: ParseError<I::Token, I::Range, I::Position>,
+{
+    attempt(string(x).skip(period()))
+        .skip(string(STATS).skip(equals()))
+        .map(|x| Param(x.to_string()))
+        .message("while getting non final param")
```
[the remark this answers is not in the document]
Comment (RDruon) [reply src/mds/mds_parser.rs]: I can't use `format!` here, I changed the error to `while parsing 'mds_suffix'`/`while parsing 'oss_suffix'`

File: src/stats_parser.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jgrund) [reply src/stats_parser.rs:112]: Bleh, ok.

File: src/mds/mds_parser.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jgrund) [reply src/mds/mds_parser.rs]: I wonder if adding static strings together produces a new static string

File: src/stats_parser.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (RDruon) [reply src/stats_parser.rs:112]: I wonder how many more might produce this kind of output

File: src/mds/mds_parser.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (RDruon) [reply src/mds/mds_parser.rs]: Looks like no:
```
[rdruon@c202 lustre-collector]$ cargo check
    Checking lustre_collector v0.7.3 (/home/rdruon/lustre-collector)
error[E0369]: cannot add `&str` to `&str`
  --> src/mds/mds_parser.rs:64:46
   |
64 |         .message("while parsing MDS suffix " + x)
   |                  --------------------------- ^ - &str
   |                  |                           |
   |                  |                           `+` cannot be used to concatenate two `&str` strings
   |                  &str
   |
   = note: string concatenation requires an owned `String` on the left
```

[the remark this answers is not in the document]
Comment (RDruon) [reply src/mds/mds_parser.rs]: https://crates.io/crates/const_concat/0.1.0 seems to work

[the remark this answers is not in the document]
Comment (jgrund) [reply src/mds/mds_parser.rs]: That's a shame. seems like a compiler shortcoming

File: src/mds/mds_parser.rs [hunk 3]
```diff
@@ -1,179 +1,100 @@
-// Copyright (c) 2021 DDN. All rights reserved.
+// Copyright (c) 2023 DDN. All rights reserved.
```
Comment (jgrund) [inline src/mds/mds_parser.rs:1]:
```suggestion
// Copyright (c) 2024 DDN. All rights reserved.
```

File: src/mds/mdt_parser.rs [hunk 4]
```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2023 DDN. All rights reserved.
```
Comment (jgrund) [inline src/mds/mdt_parser.rs:1]:
```suggestion
// Copyright (c) 2024 DDN. All rights reserved.
```

**Review by jgrund — APPROVED** [review]

**Pull request merged**